### PR TITLE
feat: added ontology-aware tissue filter. #2541.

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
- #2541 

### Reviewers
**Functional:** 
Draft.

---

## Changes
- Added ontology-aware tissue filter (behind filter feature flag). 

## Notes for Reviewer
- The ontology-aware tissue filter displays top-level tissues only.
- Refactored `FilterPanel/x` naming convention to `FilterView/x` to better reflect possible UI states of ontology-aware filters (e.g. development stage has multiple panels where as tissue does not).
- Generalized existing ontology-related naming/functionality.

### CODE REVIEW

File: frontend/src/components/common/Filter/components/FilterViews/components/FilterView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * Returns filtered ontology tree view values where value label includes search value.
+ * @param values - Ontology tree view values.
+ * @param searchValue - Search string to filters category values.
+ * @returns array of ontology tree view values filtered by the given search value.
+ */
+function filterViewValues(
+  values: OntologyCategoryTreeNodeView[] = [],
+  searchValue: string
+  // accValues: OntologyCategoryTreeNodeView[] = []
```
Comment (tihuan) [inline frontend/src/components/common/Filter/components/FilterViews/components/FilterView/index.tsx:128]: dead code 🙆‍♂️ ?

File: frontend/src/components/common/Filter/components/FilterViews/components/FilterViewList/style.ts [hunk 2]
```diff
@@ -27,3 +29,17 @@ export const useFilterPanelListStyles = makeStyles({
   },
 });
 /* eslint-enable sort-keys -- ignore object key order for style objects */
+
+export const NoMatches = styled(ListItem)`
+  && {
```
Comment (tihuan) [inline frontend/src/components/common/Filter/components/FilterViews/components/FilterViewList/style.ts:34]:
```suggestion
  /* Remove && after updating SDS version that has this commit https://github.com/chanzuckerberg/sci-components/pull/201 */
  && {
```

**Review by tihuan — APPROVED** [review]
Woohoo! Another great PR 🚀 🤩 

Thanks so much, both!!

File: frontend/src/components/common/Filter/components/FilterViews/components/FilterView/index.tsx (hunk 1 above)
Comment (frano-m) [reply frontend/src/components/common/Filter/components/FilterViews/components/FilterView/index.tsx:128]: 😶‍🌫️ - very dead ...

**Pull request merged**